lms/djangoapps/verify_student/views.py

```diff
@@ -855,6 +855,46 @@ def submit_photos_for_verification(request):
     return HttpResponse(200)
 
 
+def _get_location_id(photo_verification):
+    try:
+        ver_status = VerificationStatus.objects.filter(checkpoint__photo_verification=photo_verification).latest()
+        return ver_status.location_id
+    except VerificationStatus.ObjectDoesNotExist:
+        return ""
+
+
+def _get_user_attempts(course_key, user_id, related_assessment):
+
+    return VerificationStatus.objects.filter(
+        user_id=user_id,
+        checkpoint__course_id=course_key,
+        checkpoint__checkpoint_name=related_assessment,
+        status="submitted"
+    ).count()
+
+
+def _send_email(course_key, user_id, relates_assessment, photo_verification, status):
+    try:
+        user_attempts = _get_user_attempts(course_key, user_id, relates_assessment)
+        location_id = _get_location_id(photo_verification)
+        usage_key = UsageKey.from_string(location_id)
+        ver_block = modulestore().get_item(usage_key)
+        allowed_attempts = ver_block.attempts
+
+        if status == "approved":
+            # TODO: Render the template for success
+            text = "The verification has been approved"
+        if status in ["denied", "error"]:
+            # TODO: Render the template for failure
+            text = "The verification failed"
+            left_attempts = allowed_attempts - user_attempts
+            current_date = datetime.now()
+            # verification_open = ver_block.start_date <= current_date <= ver_block.end_date
```

Comment:
One really important thing to be aware of here: it's possible for these dates to be `None`:

* If `start` is None, the XBlock doesn't have a start date (it was open as soon as it was published)
* If `end` is None, the XBlock doesn't have an end date.  Once it's released, it's always open.